src/main/java/com/icthh/xm/ms/entity/repository/search/XmEntitySearchRepositoryImpl.java

```diff
@@ -0,0 +1,199 @@
+        return elasticsearchOperations.queryForPage(query, getEntityClass());
+    }
+
+    @Override
+    public Page<XmEntity> findAll(Pageable pageable) {
+        SearchQuery query = new NativeSearchQueryBuilder()
+            .withQuery(matchAllQuery())
+            .withIndices(elasticsearchOperations.getIndexName())
+//            .withTypes(ElasticsearchTemplateWrapper.INDEX_QUERY_TYPE) TODO-IMPL: Removed in 8.14v
+            .withPageable(pageable)
+            .build();
+        return elasticsearchOperations.queryForPage(query, getEntityClass());
+    }
+
+    @Override
+    public <S extends XmEntity> S save(S entity) {
+        String indexName = elasticsearchOperations.composeIndexName(tenantContextHolder.getTenantKey());
+        elasticsearchOperations.index(createIndexQuery(indexName, entity));
+        return entity;
+    }
+
+    @Override
+    public <S extends XmEntity> Iterable<S> saveAll(Iterable<S> entities) {
+        String indexName = elasticsearchOperations.composeIndexName(tenantContextHolder.getTenantKey());
+
+        List<IndexQuery> queries = new ArrayList<>();
+        for (S s : entities) {
+            queries.add(createIndexQuery(indexName, s));
+        }
+
+        elasticsearchOperations.bulkIndex(queries);
+        elasticsearchOperations.refresh(indexName);
+        return entities;
+    }
+
+    private <S extends XmEntity> IndexQuery createIndexQuery(String indexName, S entity) {
+        IndexQuery query = new IndexQuery();
+        query.setObject(entity);
+        query.setId(String.valueOf(entity.getId()));
+        query.setType(ElasticsearchOperations.INDEX_QUERY_TYPE);
+        query.setIndexName(indexName);
+//        query.setVersion(extractVersionFromBean(entity)); // TODO
+//        query.setParentId(extractParentIdFromBean(entity)); // TODO
+        return query;
+    }
+
+
+    @Override
+    public Optional<XmEntity> findById(Long id) {
+        GetQuery query = new GetQuery();
+        query.setId(String.valueOf(id));
+        return Optional.ofNullable(elasticsearchOperations.queryForObject(query, getEntityClass()));
+    }
+
+    @Override
+    public boolean existsById(Long id) {
+        return findById(id).isPresent();
+    }
+
+    @Override
+    public Iterable<XmEntity> findAll() {
+        int itemCount = (int) this.count();
+        if (itemCount == 0) {
+            return new PageImpl<>(Collections.<XmEntity> emptyList());
+        }
+        return this.findAll(PageRequest.of(0, Math.max(1, itemCount)));
+    }
+
+    @Override
+    public Iterable<XmEntity> findAllById(Iterable<Long> longs) {
+        throw new UnsupportedOperationException("Not implemented");
+    }
+
+    @Override
+    public long count() {
+        SearchQuery query = new NativeSearchQueryBuilder()
+            .withQuery(matchAllQuery())
+            .withIndices(elasticsearchOperations.getIndexName())
+//            .withTypes(ElasticsearchTemplateWrapper.INDEX_QUERY_TYPE)
```

Comment:
@osiievska23 check and remove if it not needed

Reply:
done